### DISCUSSION TITLE
MPI Rattle Segfaults

### DISCUSSION THREAD

**Discussion by johnemmanuelbond** [glotzerlab/hoomd-blue#1742] [Support] [answered]
Hi hoomd devs,

Since version 3.9 I have been struggling getting domain decomposition to work with molecular dynamics sims of particles confined to a surface using rattle. Until recently I was just ineffeciently running sims on one core, but with recent changes to my university's supercomputing cluster I need to figure this out before I waste more resources than just my own time.

I've attached below a folder with a simple script that recreates the segfault errors in hoomd version 4.0.1. It simply runs md.methods.rattle.Brownian on a spherical surface with some circular DLVO particles. When I run it on one core it works, when I try to run it on 8 it always segfaults. There's more info in the readme, but any help resolving this issue would be greatly appreciated.

Thanks!
\- Jack Bond

[hoomd_rattle_mpi_segfault.zip](https://github.com/glotzerlab/hoomd-blue/files/14593744/hoomd_rattle_mpi_segfault.zip)

**Comment by joaander** [comment, marked as answer]
Thanks for the complete test case. I was able to reproduce the error and fix it: #1748.

**Reply by johnemmanuelbond** [reply]
Excellent, Thanks! Is this in a release version I can clone and build?

**Reply by joaander** [reply]
Not at this time. It will be included in the 4.7.0 release.
